pkg/flowcli/project/account.go

```diff
@@ -32,6 +32,18 @@ func (a *Account) SetDefaultKey(key AccountKey) {
 	a.keys[0] = key
 }
 
+func (a *Account) ValidateKey() error {
+	if a.DefaultKey().Type() == config.KeyTypeGoogleKMS {
+		resourceID := a.DefaultKey().ToConfig().Context[config.KMSContextField]
+		err := util.GcloudApplicationSignin(resourceID)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Would it make sense to move this validation into the key struct?

```suggestion
	return a.DefaultKey().Validate()
```

Reply:
💯